tests/test_simulator.py

```diff
@@ -0,0 +1,67 @@
+import numpy
+
+from model import state, action
+import simulator
+
+
+class SingleDimStateSpace(state.StateSpace):
+    def __init__(self, len):
```

Comment:
Although `len` isn't a keyword, it's an important built-in function that can be confusing if you override it with a local variable. I'd call this `length` or `size`